src/app/modules/spcp/spcp.util.ts

```diff
@@ -210,6 +220,22 @@ export const createCorppassParsedResponses = (
   ]
 }
 
+/**
+ * Validates that a form is a SPCP form with an e-service ID
+ * @param form Form to validate
+ */
+export const validateSpcpForm = (
+  form: IFormSchema | IPopulatedForm,
+): Result<ISpcpForm, SpcpNoESrvcIdError | SpcpAuthTypeError> => {
+  if (!form.esrvcId) {
+    return err(new SpcpNoESrvcIdError())
+  }
+  if (form.authType !== AuthType.SP && form.authType !== AuthType.CP) {
```

Comment:
With the composite type above, if we make this into a type guard instead we can get better typings and remove the need for type casting

```ts
const isSpcpForm = <F extends IFormSchema>(form: F): form is SpcpForm<F> => {
  return [AuthType.SP, AuthType.CP].includes(form.authType)
}
```

Reply:
good suggestion! will also do teh same for `MyInfo` and look to preserve type information through typeguards in the future